woocommerce/woocommerce-android
# Connecting products

Part of: #12998

### Description
This PR connects the product card with real data. To achieve this, the GetShippableItems use case is used to get valid shippable items, and the `WooShippingLabelCreationViewModel` is updated to expose the results. 

I decided to include two different classes to represent shippable items: `ShippableItemModel` and `ShippableItemUI`. ShippableItemModel will be useful for calculating totals, split shipments, etc., while  ShippableItemUI will be the UI representation of the product, including currency format and dimensions.

### Testing information

1. Create an Order with the processing status on a store that can create shipping labels
2. Tap on Orders
3. Tap on the order created on step 1
4. Tap on create shipping labels
5. Check that the current shippable products are displayed in the product card

### The tests that have been performed

1. Checking the UI works as expected

### Images/gif

https://github.com/user-attachments/assets/206f77cc-3a8e-4659-a9d0-311b1c63876e



## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
